giskard/ml_worker/exceptions/giskard_exception.py

```diff
@@ -1,3 +1,48 @@
+from typing import Optional, Tuple
+import platform
+
+
 class GiskardException(Exception):
     def __init__(self, *args: object) -> None:
         super().__init__(*args)
+
+
+class GiskardPythonEnvException(GiskardException):
+    def __init__(self, *args: object) -> None:
+        super().__init__(*args)
+
+
+class GiskardPythonVerException(GiskardPythonEnvException):
+    def __init__(self, cls_name, e, required_py_ver, *args: object) -> None:
+        super().__init__(
+            f"Failed to load '{cls_name}' due to {e.__class__.__name__}.\n"
+            f"Make sure you are loading it in the environment with matched Python version (required {required_py_ver}, loading with {platform.python_version()}).",
+            *args,
+        )
+
+
+class GiskardPythonDepException(GiskardPythonEnvException):
+    def __init__(self, cls_name, e, *args: object) -> None:
+        super().__init__(
+            f"Failed to load '{cls_name}' due to {e.__class__.__name__}.\n"
+            "Make sure you are loading it in the environment with matched dependencies.",
+            *args,
+        )
+
+
+def python_env_exception_helper(cls_name, e: Exception, required_py_ver: Optional[Tuple[str, str, str]] = None):
+    if required_py_ver is not None and required_py_ver[:2] != platform.python_version_tuple[:2]:
```

Comment:
I also worry about the false positive here: it's actually caused by dependency issues, but the Python version mismatch leads to this branch. @kevinmessiaen WDYT

Reply:
That's true that it'll create false positive, however for now we only use pickle so that's not a big deal. We will probably handle Python code differently once we improve the code serialization

Reply:
How about `TypeError` and `minor` version not matched?

Reply:
Well we should probably keep track of the Python version that we used to generate pickle since we used [cloudpickle](https://github.com/cloudpipe/cloudpickle)

Has stated in there readme:
![image](https://github.com/Giskard-AI/giskard/assets/114553769/e2b70442-c69c-4045-96b2-f30353f7a6a0)

Here is an explanation of why it is the case: https://github.com/cloudpipe/cloudpickle/issues/293#issuecomment-502011167

Catching `TypeError` is not reliable enough IMO, better keeping track of the version internally

Reply:
Done!